crawler/src/main.rs

```diff
@@ -375,8 +375,19 @@ async fn main() -> anyhow::Result<()> {
                         let domain_doc = domains_collection
                             .find_one(doc! { "domain": domain }, None)
                             .await?
-                            .ok_or_else(|| anyhow::anyhow!("missing domain for {url}"))?;
-
+                            .ok_or_else(|| anyhow::anyhow!("missing domain for {url}"));
+
+                            let domain_doc = match domain_doc {
+                                Ok(doc) => doc,
+                                Err(err) => {
+                                    let id = match document.get_object_id("_id") {
+                                        Ok(id) => id.to_hex(),
+                                        Err(_) => "<unknown>".into(),
+                                    };
+                                    erroneous_docs.insert(id, err);
+                                    continue;
+                                }
+                            };
```

Comment:
Small nitpick (since you'll have to change this anyway): please add a blank line between `};` and the next line.